### ISSUE TITLE
transpile: Fresh statics conflict with locals defined earlier

### ISSUE THREAD

**Issue by Rua** [immunant/c2rust#1994]
I generate both a static variable and a local variable that each generate a fresh variable. The static is defined after the local:

```c
void foo(void) {
    int *local_pointer = &(int){ 42 };
}

int *static_pointer = &(int){ 42 };
```

```rust
#[no_mangle]
pub unsafe extern "C" fn foo() {
    let mut c2rust_lvalue: ::core::ffi::c_int = 42 as ::core::ffi::c_int;
    let mut local_pointer: *mut ::core::ffi::c_int = &raw mut c2rust_lvalue;
}
static mut c2rust_lvalue: ::core::ffi::c_int = 42;
#[no_mangle]
pub static mut static_pointer: *mut ::core::ffi::c_int = unsafe {
    &raw const c2rust_lvalue as *mut ::core::ffi::c_int
};
```

This triggers a compile error from Rust:
> let bindings cannot shadow statics

This doesn't happen if the order of the definitions is reversed, then the local `c2rust_lvalue` gets renamed.
